### PR TITLE
feat: support Checks API, and so GitHub Actions badge

### LINKED ISSUE #304: Github Actions (CI/CD) badge
Seems like Github already have a badge at 

```
https://github.com/<user>/<repo>/workflows/<workflow name>/badge.svg
```

https://github.com/actions/toolkit/workflows/Main%20workflow/badge.svg

But as we know, our badges are better :P

### LINKED ISSUE #310: GitHub actions status incorrect?
I think something is up with GitHub Actions statuses but I can’t tell what…

I have a “[Tests](https://github.com/brandonweiss/charge/actions)” action. The badge should be this:

https://badgen.net/github/status/brandonweiss/charge/master/Tests

It was working but now it’s returning “unknown”. Strangely, here’s what happens if I lop off the action name:

https://badgen.net/github/status/brandonweiss/charge/master

It says “pending” even though nothing is pending as far as I can tell 🤔

I can’t tell if this an issue on the Badgen side or if GitHub broke something?

### PR DESCRIPTION
**Pull request by tunnckoCore**
fixes #304 & fixes #310

```
/github/checks/:owner/:repo/:ref?
/github/checks/tunnckoCore/opensource?label=build&icon=github
```

https://badgennet.charlike-oss.now.sh/github/checks/tunnckoCore/opensource

[](https://github.com/tunnckoCore/opensource/actions)
[](https://github.com/tunnckoCore/opensource/actions)

The example is pointing to my monorepo. If there are better examples, okay.
I'm planning to use GHA non-stop anyway. Also, the repo is not some dumb one, but my main focus and moving everything there.

### CODE REVIEW

File: endpoints/github.ts [hunk 1]
```diff
@@ -111,17 +113,29 @@ const statesColor = {
   unknown: 'grey'
 }
 
-function combineState (states: Array<any>) {
+function combined (states: Array<any>, key?: string) {
```
Comment (amio) [inline endpoints/github.ts]: How about this signature?

```js
function combineStates (states: Array<any>, stateKey = 'state') {}
```

File: endpoints/github.ts [hunk 2]
```diff
@@ -111,17 +113,29 @@ const statesColor = {
...
+  const k = stateKey || 'state'
   if (states.length === 0) return 'unknown'
-  if (states.find(x => x.state === 'error')) return 'error'
-  if (states.find(x => x.state === 'failure')) return 'failure'
-  if (states.find(x => x.state === 'pending')) return 'pending'
-  if (states.every(x => x.state === 'success')) return 'success'
+  if (states.find(x => x[k] === 'error')) return 'error'
+  if (states.find(x => x[k] === 'failure')) return 'failure'
+  if (states.find(x => x[k] === 'pending')) return 'pending'
+  if (states.every(x => x[k] === 'success')) return 'success'
```
Comment (amio) [inline endpoints/github.ts]: I mean, we can give a default value to `k` in the arguments:

```js
function combined (states: Array<any>, stateKey = 'state') {
  if (states.length === 0) return 'unknown'
  if (states.find(x => x[stateKey] === 'error')) return 'error'
  if (states.find(x => x[stateKey] === 'failure')) return 'failure'
  if (states.find(x => x[stateKey] === 'pending')) return 'pending'
  if (states.every(x => x[stateKey] === 'success')) return 'success'
```

just like you did in the `checks()` function:

```js
async function checks ({ owner, repo, ref = 'master'}: Args) {
```

Comment (tunnckoCore) [reply endpoints/github.ts]: Yea we can.

Comment (amio) [reply endpoints/github.ts]: If it has a default value, then it is optional 😉

image

https://www.typescriptlang.org/play/index.html?ssl=3&ssc=1&pln=3&pc=13#code/GYVwdgxgLglg9mABBOBbARjMBTAJgCgGcoBDKbQgLkQEEAnOkgTwB4SwmA+AGkWLOwBpbE0QBeRAHJ+5SQEpEAb0QBfALAAoTSgxY8+ANoBdOUA

**Comment by amio** [comment]
@tunnckoCore Thanks!

**Comment by tunnckoCore** [comment]
Awesome! :heart_eyes_cat:

**Pull request merged**